fenix/app/src/main/java/org/mozilla/fenix/onboarding/view/OnboardingScreen.kt

```diff
@@ -65,16 +64,21 @@ fun OnboardingScreen(
     pagesToDisplay: List<OnboardingPageUiData>,
     onMakeFirefoxDefaultClick: () -> Unit,
     onSkipDefaultClick: () -> Unit,
-    onPrivacyPolicyClick: (url: String) -> Unit,
     onSignInButtonClick: () -> Unit,
     onSkipSignInClick: () -> Unit,
     onNotificationPermissionButtonClick: () -> Unit,
     onSkipNotificationClick: () -> Unit,
     onAddFirefoxWidgetClick: () -> Unit,
     onSkipFirefoxWidgetClick: () -> Unit,
-    onFinish: (pageType: OnboardingPageUiData) -> Unit,
+    onFinish: (pageType: OnboardingPageUiData?) -> Unit,
     onImpression: (pageType: OnboardingPageUiData) -> Unit,
 ) {
+    /* do not continue if there's no onboarding pages to display */
+    if (pagesToDisplay.isEmpty()) {
+        onFinish(null)
+        return
+    }
```

Comment:
I think this should be in the Fragment. We create pagesToDisplay list there and we can check if it's empty `onCreate` and just finish there. 

A tip 💡  we don't need this in compose but if we did, we should be doing this kind of check in an [effect handler](https://developer.android.com/jetpack/compose/side-effects) in composable like a `LaunchedEffect`, without returning.

Reply:
Good idea, I'll move the check.  Thanks